### PR TITLE
Hotfix: Update `@bolt/core` Base Component Slot Check

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
http://vjira2:8080/browse/WWWD-2628

## Summary
Workaround to address extreme edge case of dynamically injected, server-side pre-rendered buttons not hydrating properly during the component's async rendering process.

## How to test
Compare test examples included in this PR -- notably the server-side rendered examples with the added delay -- in Firefox, IE 11 and Edge.

- [Content-injection w/ client-side rendering:](https://boltdesignsystem-psyvnhjclw.now.sh/pattern-lab/?p=components-personalized-card-example--client-rendered-tests)
- [Content-injection w/ server-side rendering:](https://boltdesignsystem-psyvnhjclw.now.sh/pattern-lab/?p=components-personalized-card-example--server-rendered-tests)
> For reference, the main issue being addressed here should be most visible when looking at the server-side pre-rendered examples -- particularly when a long (and sometimes short) delay is encountered

### CODE REVIEW

**Comment by sghoweri** [comment]
^ there's something funny going on with our build script that creates the feature-branch specific URL alias during the Travis build -- here's the now.sh url for reference (while I go troubleshoot that): https://boltdesignsystem-psyvnhjclw.now.sh

File: packages/core/renderers/bolt-base.js [hunk 1]
```diff
@@ -55,6 +55,11 @@ export function BoltBase(Base = HTMLElement) {
       // this ensures that things work as expected, even when a component gets removed / re-added to the page
       this.setupShadow();
 
+      // @todo: rework to disable this extra check here (used for demoing before/after behavior), unless a specific feature flag is used (debug mode?)
+      if (!this.closest('.js-disable-extra-slot-check')) {
```
Comment (remydenton) [inline packages/core/renderers/bolt-base.js]: So every component that uses `BoltBase`, including everything used in production, now has to also run this `.closest()` check when rendering?  I think it's a stretch to justify that just for testing.  I'd feel better if we created a follow up ticket so this doesn't just languish here.

File: apps/pattern-lab/src/index.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import './_patterns/02-components/card/__tests__/';
+
```
Comment (remydenton) [inline apps/pattern-lab/src/index.js]: This is temporary, right?  There must be a better way to load tests without hard-coding a path.

**Comment by remydenton** [comment]
Glad you got to the bottom of this!  See my inline comments as well as the following:

- Since this is a hotfix, we should change the base branch to `release/2.x`.  That way, we're not forced to release all the other stuff on the working master branch and potentially introduce more regressions at the 11th hour.

- Having the tests to demonstrate the fix (and help prevent it from happening again) is excellent.  However, I really don't think they're in the right place.  Right now, they're side-by-side with our Bolt usage documentation for end users in Pattern Lab, which is confusing for everyone (end users and Bolt developers alike).
I honestly think we should be introducing testing as a separate PR.  That would include:
    - File organization and naming for tests.  Maybe even a separate app for tests?
    - A polished example of a test
    - Documentation in the wiki for at least the basics of adding a test

I'm guessing all that is coming in http://vjira2:8080/browse/BDS-714--  can you confirm @sghoweri?  If the answer is "yes", we can merge this in on the condition this all gets ironed out soon and not left to collect dust.

File: apps/pattern-lab/src/index.js (hunk 2 above)
Comment (sghoweri) [reply apps/pattern-lab/src/index.js]: Yep, very much so. This won't be necessary once my next batch of testing coverage-related updates is in.
